src/api/FileCollaborators.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @flow
+ * @file Helper for the box collaborators API
+ * @author Box
+ */
+
+import MarkerBasedAPI from './MarkerBasedAPI';
+
+class FileCollaborators extends MarkerBasedAPI {
+    /**
+     * API URL for comments
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} base url for files
+     */
+    getUrl(id: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+        return `${this.getBaseApiUrl()}/files/${id}/collaborators`;
+    }
+
+    /**
+     * Generic success handler
+     *
+     * @param {Object} data the response data
+     */
+    successHandler = (data: any): void => {
+        const { entries } = data;
+        const collaborators = entries.map((collab) => {
+            const { id, name, login } = collab;
+            return {
+                id,
+                name,
+                item: { ...collab, email: login }
+            };
+        });
+
+        if (!this.isDestroyed() && typeof this.successCallback === 'function') {
```

Comment:
Probably just at the starting of this function you can add the guard, so that you don't even bother doing the above processing.

```if (this.isDestroyed() ...) { return; }```

Also, will successCallback ever be not a function in this case?

Reply:
Not sure if it will ever not be a function. The overridden function checked for that case